service/dap/server.go

```diff
@@ -1511,10 +1555,105 @@ func (s *Server) onReverseContinueRequest(request *dap.ReverseContinueRequest) {
 	s.sendNotYetImplementedErrorResponse(request.Request)
 }
 
-// onSetVariableRequest sends a not-yet-implemented error response.
-// Capability 'supportsSetVariable' is not set 'initialize' response.
-func (s *Server) onSetVariableRequest(request *dap.SetVariableRequest) { // TODO V0
-	s.sendNotYetImplementedErrorResponse(request.Request)
+func variableByName(v *proc.Variable, name string) (*proc.Variable, bool) {
+	if v == nil {
+		return nil, false
+	}
+	if v.Name == name {
+		return v, true
+	}
+
+	for _, child := range v.Children {
+		if child.Name == name {
+			return &child, true
+		}
+		if v, ok := variableByName(&child, name); ok {
```

Comment:
I don't understand why this recursion is needed.

Reply:
It's from my misunderstanding of how DAP client would access nested fields. Removed.